cuhe/CMakeLists.txt

```diff
@@ -0,0 +1,36 @@
+CMAKE_MINIMUM_REQUIRED(VERSION 2.8)
+PROJECT(cuhe)
+
+SET(GPU_ARCH "50" CACHE STRING "Cuda architecture number, one of ('20', '30', '32', '35', '37', '50', '52', '53')")
+SET(GCC_CUDA_VERISON "4.9" CACHE STRING "GCC compiler to be used with cuda")
+
+SET(CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake)
+
+FIND_PACKAGE(NTL REQUIRED)
+FIND_PACKAGE(OpenMP REQUIRED)
+FIND_PACKAGE(CUDA REQUIRED)
+
+INCLUDE(FindCUDA)
+
+FILE(GLOB SOURCES "*.cu" "*.h")
+
+LIST(APPEND CMAKE_CXX_FLAGS ${OpenMP_CXX_FLAGS})
+LIST(APPEND CMAKE_CXX_FLAGS "-m64 -std=c++11 -O3 -ffast-math -Wall")
+
+LIST(APPEND CUDA_NVCC_FLAGS "-ccbin gcc-${GCC_CUDA_VERISON} -std=c++11 -gencode arch=compute_${GPU_ARCH},code=sm_${GPU_ARCH}")
```

Comment:
I t depends on the version of gcc you will use within cuda probably you previously linked your version within your cuda installation (as I did having making this working).
Probably "4.8.4" will be available on "4.8" if this doesn't work please let me know the exact version of your libraries and I will try to repro the problem.

What do you think about using cmake?